activerecord/lib/active_record/counter_cache.rb

```diff
@@ -105,7 +105,8 @@ def update_counters(id, counters)
         end
 
         if touch
-          updates << sanitize_sql_for_assignment(touch_updates(touch))
+          touch_updates = touch_updates(touch)
+          updates << sanitize_sql_for_assignment(touch_updates) unless touch_updates.empty?
```

Comment:
Why do we need this change? Presumably `sanitize_sql_for_assignment` returns an empty string that's safe to append?